### PR TITLE
Update for 10.3.7 + ChannelSupport

### PR DESCRIPTION
**Pull request by Nickbert7**

### CODE REVIEW

File: Pushbullet/Api/ServerApiEntryPoints.cs [hunk 1]
```diff
@@ -24,11 +25,13 @@ public class TestNotification : IReturnVoid
...
     {
         private readonly IHttpClient _httpClient;
+		private readonly IJsonSerializer _jsonSerializer;
         private readonly ILogger _logger;
 
-        public ServerApiEndpoints(ILogger logger, IHttpClient httpClient)
+        public ServerApiEndpoints(ILogger logger, IJsonSerializer jsonSerializer, IHttpClient httpClient)
         {
               _logger = logger;
+			  _jsonSerializer = jsonSerializer;
```
Comment (cvium) [inline Pushbullet/Api/ServerApiEntryPoints.cs]: You're using tabs it seems. We use 4 spaces.
```suggestion
              _jsonSerializer = jsonSerializer;
```

File: Pushbullet/Api/ServerApiEntryPoints.cs [hunk 2]
```diff
@@ -61,16 +64,19 @@ public async Task PostAsync(TestNotification request)
...
-            _httpRequest.Url = "https://api.pushbullet.com/v2/pushes";
-
-            _httpRequest.SetPostData(parameters);
-
-            using (await _httpClient.Post(_httpRequest).ConfigureAwait(false))
-            {
-
-            }
+            var requestOptions = new HttpRequestOptions
+			{
```
Comment (cvium) [inline Pushbullet/Api/ServerApiEntryPoints.cs]: Indentation should be spaces.

Comment (Nickbert7) [reply Pushbullet/Api/ServerApiEntryPoints.cs]: Ok, should be fixed now

**Review by cvium — DISMISSED** [review]

File: Pushbullet/Configuration/PluginConfiguration.cs [hunk 3]
```diff
@@ -19,6 +19,7 @@ public class PushbulletOptions
         public Boolean Enabled { get; set; }
         public String Token { get; set; }
         public String DeviceId { get; set; }
+        public String Channel { get; set; }
```
Comment (cvium) [inline Pushbullet/Configuration/PluginConfiguration.cs]: It should be lowercase `string` and `bool` for these properties.
```suggestion
        public string Channel { get; set; }
```

File: Pushbullet/Notifier.cs [hunk 4]
```diff
@@ -48,26 +51,30 @@ public async Task SendNotification(UserNotification request, CancellationToken c
             var parameters = new Dictionary<string, string>
                 {
                    // {"device_iden", options.DeviceId},
+                    {"channel_tag", options.Channel},
```
Comment (cvium) [inline Pushbullet/Notifier.cs:54]: Does this work if it's null or empty?

Comment (Nickbert7) [reply Pushbullet/Notifier.cs:54]: Yes, I have tested it.
When the string is empty, I get the notifications on all my devices, when I enter the cannel tag, it is pushed to the channel.

**Review by cvium — DISMISSED** [review]

**Comment by joshuaboniface** [comment]
Will this work the same between 10.3.7 and 10.4.0? If so, let's :shipit: , if not let's wait like the other plugin for 10.4.0 and rebuild then.

**Comment by Nickbert7** [comment]
It is currently only woking for 10.3.7 ant not for 10.4.0

**Comment by Nickbert7** [comment]
OK, it was an easy fix, now it is working for 10.3.7 and 10.4.0

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**